worker.go

```diff
@@ -275,12 +275,13 @@ func (w *Worker) Flush() WorkerMetrics {
 	// This is a critical spot. The worker can't process metrics while this
 	// mutex is held! So we try and minimize it by copying the maps of values
 	// and assigning new ones.
+	wm := NewWorkerMetrics()
 	w.mutex.Lock()
```

Comment:
Seems like we could make the worker metrics into atomics and avoid this locking all together?

Reply:
We could, i actually did a bunch of that locally already - was gonna delay to another PR, heh

Reply:
Np, you can skip it and do it in another PR or not do it at all.